.github/workflows/ci.yml

```diff
@@ -0,0 +1,112 @@
+name: continuous-integration
+on: [push]
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        py: [ 3.6, 3.7, 3.8, 3.9 ]
+        CC: [ gcc ]
+
+        # Check macos too
+        include:
+          - os: macos-latest
+            py: 3.7
+
+    steps:
+        - uses: actions/checkout@v2
+
+        - name: Set up Python ${{ matrix.py }}
+          uses: actions/setup-python@v2
+          with:
+            python-version: ${{ matrix.py }}
+
+        - name: Install GFortran macOS
+          if: matrix.os == 'macos-latest'
+          run: |
+            brew cask install gfortran || true
+
+        - name: Install GSL, FFTW, SWIG (macos)
+          if: matrix.os == 'macos-latest'
+          run: |
+            brew install fftw || true
+            brew install gsl || true
+            brew install swig || true
+
+        - name: Install GSL, FFTW (linux)
+          if: matrix.os == 'ubuntu-latest'
+          run: |
+            sudo -H apt-get install libgsl-dev libfftw3-dev
+
+        - name: Install python dependencies
+          run: |
+            python -m pip install -U pip
+            pip install -U numpy
+            pip install -U scipy
+            pip install -U pytest
+            pip install -U pytest-cov
+            pip install -U fast-pt
+            pip install -U camb
+            pip install -U isitgr
+            pip install -U cython
+            pip install -U flake8
+
+        - name: Install CLASS (linux)
+          if: matrix.os == 'ubuntu-latest'
+          run: |
+            git clone -b v2.7.2 --single-branch --depth 1 "https://github.com/lesgourg/class_public.git"
+            cd class_public
+            make
+
+        - name: Install CLASS (macos)
+          if: matrix.os == 'macos-latest'
+          run: |
+            git clone -b v2.7.2 --single-branch --depth 1 "https://github.com/lesgourg/class_public.git"
+            cd class_public
+            make CC=gcc-9
+
+        - name: Flake
+          run: |
+            flake8 pyccl
+            flake8 --exclude=data benchmarks
+
+        - name: Build CCL
+          run: |
+            python setup.py build
+            python setup.py develop
+
+        - name: C unit tests
+          run: |
+            cd build
+            make -j4
+            CLASS_PARAM_DIR=./extern/share/class/ OMP_NUM_THREADS=2 ./check_ccl
+
+        - name: Python unit tests
+          run: OMP_NUM_THREADS=2 pytest -vv pyccl --cov=pyccl
+
+        - name: Benchmarks
+          run: OMP_NUM_THREADS=2 pytest -vv benchmarks --cov=pyccl --cov-append
+
+        - name: Coveralls
+          env:
+            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          run: |
+            pip install -U coveralls
+            coveralls
```

Comment:
Maybe consider adding a cache?  I don't have any experience yet how important it is for GHA.  But it sped things up a lot in Travis to cache pip and brew directories.  Here is what that would look like I think:
```
           - name: Cache pip, brew
              uses: actions/cache@v2
              with:
                path: |
                  ~/.cache/pip
                  $HOME/Library/Caches/Homebrew
                  /usr/local/Cellar
                key: ${{ runner.os }}-${{ matrix.py }}-cache
                restore-keys: |
                    ${{ runner.os }}-${{ matrix.py }}-cache
                    ${{ runner.os }}-
```

Reply:
OK, caching didn't seem to have a huge impact, but I've left it in just in case (doesn't hurt to have it).